### PR TITLE
utils.pwsh: Support all Get-FileHash algorithms

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

Get-FileHash supports the following hash algorithms:
 * SHA1
 * SHA256
 * SHA384
 * SHA512
 * MD5

Let's support all of those instead of hard-coding SHA256 only for downloading files via Invoke-SafeWebRequest.

### Motivation and Context

More easily check hashsums if projects did not provide SHA256, but did provide hashes from another supported algorithm.

### How Has This Been Tested?

Manually tested locally.

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by notr1ch** [comment]
SHA1 and MD5 are considered broken / insecure, so I would not recommend using those for verifying packages. If those are the only published hashes, then I think it's safer if we verify the package manually and create a SHA256 hash ourselves.

File: utils.pwsh/Invoke-SafeWebRequest.ps1 [hunk 1]
```diff
@@ -39,8 +39,15 @@ function Invoke-SafeWebRequest {
     try {
         $HashData = Import-Clixml $HashFile
 
+        $SupportedAlgorithms = @("SHA1", "SHA256", "SHA384", "SHA512", "MD5")
```
[the remark this answers is not in the document]
Comment (RytoEX) [reply utils.pwsh/Invoke-SafeWebRequest.ps1:42]: I'll see what I can do. Though, the behavior in this patch is identical to the current behavior if an XML file specified a hash algorithm other than SHA256.

[the remark this answers is not in the document]
Comment (PatTheMav) [reply utils.pwsh/Invoke-SafeWebRequest.ps1:42]: Wouldn't it make sense to detect that algorithm (given that we import a Powershell object) and use it for the comparison automatically and just fail if (for some reason) it's not on the list of supported ones?

[the remark this answers is not in the document]
Comment (RytoEX) [reply utils.pwsh/Invoke-SafeWebRequest.ps1:42]: Ideally, yes, probably. At the time, I just went with whatever most closely followed the current behavior to make the changes minimal.

**Comment by RytoEX** [comment]
Per off-thread conversation, will merge as-is. We can make further adjustments like warning on MD5/SHA1 usage, passing through the algorithm value directly to Get-FileHash (in case Get-FileHash gains support for more algos than these 5), and therefore emitting a clearer error message when Get-FileHash receives an unsupported algorithm instead of a confusing hash mismatch.

**Pull request merged**